Makefile

```diff
@@ -53,13 +54,28 @@ $(BIN)/libbackscrub.a: $(BIN)/libbackscrub.o $(BIN)/transpose_conv_bias.o $(TFLI
 	cp -p $(TFLIBS)/libtensorflow-lite.a $@
 	ar rv $@ $(BIN)/libbackscrub.o $(BIN)/transpose_conv_bias.o
 
-$(BIN)/%.o: %.cc
-	g++ $^ ${CFLAGS} ${TFCFLAGS} -c -o $@
+# Video I/O library - this is a Linux/v4l2loopback only target for now but replaceable later..
+$(BIN)/libvideoio.a: $(BIN)/loopback.o
+	ar rv $@ $^
+
+# Compile rules for various source directories
+$(BIN)/%.o: lib/%.cc
+	g++ $< ${CFLAGS} ${TFCFLAGS} -c -o $@
+
+$(BIN)/%.o: videoio/%.cc $(TFDOWN)
+	g++ $< ${CFLAGS} ${TFCFLAGS} -c -o $@
+
+$(BIN)/%.o: app/%.cc $(TFDOWN)
+	g++ $< ${CFLAGS} ${TFCFLAGS} -c -o $@
 
 # As cloned, TFLite needs building into a static library, as per:
 # https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/make
+# we split this into download (pre-compile) and build (pre-link)
+$(TFDOWN): $(TFLITE)
+	cd $(TFLITE) && ./download_dependencies.sh
+
 $(TFLIBS)/libtensorflow-lite.a: $(TFLITE)
```

Comment:
Shouldn't this depend on `${TFDOWN}` (or better a marker file to avoid making a directory the target and thus this rule implicitly `.PHONY`).

Reply:
I needed a folder (with a datestamp) to trigger the dowload, which makes rebuilds work as expected (no download when already done - don't think directories are implicitly .PHONY?). The dependency back to `$(TFLITE)` is to trigger the `git submodue update`. Seems to all work...

Reply:
Clear from the intention. Just not sure if the `$(TFLIBS)/libtensorflow-lite.a` should better depend on `${TFDOWN}` instead to force completion of the download step before `$(TFLIBS)/libtensorflow-lite.a` is tried to be built. This is likely to work with single threaded builds, but asking to break with parallel builds.

Reply:
Good point on parallel builds, adopting your suggestion :smile: